deploy/runtools/firesim_topology_with_passes.py

```diff
@@ -268,31 +300,34 @@ def pass_perform_host_node_mapping(self):
         else:
             assert False, "IMPROPER MAPPING CONFIGURATION"
 
-    def pass_apply_default_hwconfig(self):
+    def pass_apply_default_hwconfig(self) -> None:
         """ This is the default mapping pass for hardware configurations - it
         does 3 things:
             1) If a node has a hardware config assigned (as a string), replace
-            it with the appropriate RuntimeHWConfig object.
+            it with the appropriate RuntimeHWConfig object. If it already a
+            RuntimeHWConfig object then keep it the same.
             2) If a node's hardware config is none, give it the default
             hardware config.
             3) In either case, call get_deploytriplet_for_config() once to
             make the API call and cache the result for the deploytriplet.
         """
         servers = self.firesimtopol.get_dfs_order_servers()
-        defaulthwconfig_obj = self.hwdb.get_runtimehwconfig_from_name(self.defaulthwconfig)
 
         for server in servers:
-            servhwconf = server.get_server_hardware_config()
-            if servhwconf is None:
+            hw_cfg = server.get_server_hardware_config()
```

Comment:
I personally think this code is very confusing... in some places you expected the HW_CFG to be a `str` and in others you wanted a `RuntimeHWConfig` object. This keeps the same functionality but IMO this is ugly.